### PR TITLE
Replace usd aabb

### PR DESCRIPTION
**Pull request by hang-yin**
TODO: 

- [x] fix visual aabb default case
- [x] completely remove BoundingBoxAPI

### CODE REVIEW

File: omnigibson/object_states/particle_modifier.py [hunk 1]
```diff
@@ -413,7 +413,11 @@ def check_in_adjacency_mesh(particle_positions):
             def check_overlap():
                 nonlocal valid_hit
                 valid_hit = False
-                aabb = self.link.aabb
+                try: 
+                    aabb = self.link.visual_aabb
+                except NotImplementedError:
+                    # If the link doesn't have an AABB, we can't check for overlaps
+                    return False
```
Comment (ChengshuLi) [inline omnigibson/object_states/particle_modifier.py:420]: this seems very hacky, when would this be triggered?

File: omnigibson/prims/entity_prim.py [hunk 2]
```diff
@@ -1241,11 +1237,51 @@ def aabb(self):
...
+                    continue
+                
+                position, orientation = link.get_position_orientation()
+                scale = link.scale
+                points_scaled = hull_points * scale
+                points_rotated = np.dot(T.quat2mat(orientation), points_scaled.T).T
+                points_transformed = points_rotated + position
+                points_world.append(points_transformed)
+
+            if not points_world:
```
Comment (ChengshuLi) [inline omnigibson/prims/entity_prim.py:1253]: it also looks fairly hacky, when would this trigger?

File: omnigibson/prims/rigid_prim.py [hunk 3]
```diff
@@ -605,6 +605,142 @@ def contact_reporting_enabled(self):
         """
         return self._prim.HasAPI(lazy.pxr.PhysxSchema.PhysxContactReportAPI)
 
+    def _compute_points_on_convex_hull(self, visual):
+        """
+        Returns:
+            np.ndarray: points on the convex hull of all points from child geom prims
```
Comment (ChengshuLi) [inline omnigibson/prims/rigid_prim.py:611]: this can also return None, revise docstring

File: omnigibson/prims/rigid_prim.py [hunk 4]
```diff
@@ -605,6 +605,142 @@ def contact_reporting_enabled(self):
...
+            return None
+
+        points = np.concatenate(points, axis=0)
+        
+        try:
+            hull = ConvexHull(points)
+            return points[hull.vertices, :]
+        except scipy.spatial.qhull.QhullError:
+            # Handle the case where a convex hull cannot be formed (e.g., collinear points)
+            return None
```
Comment (ChengshuLi) [inline omnigibson/prims/rigid_prim.py:631]: In this case, shouldn't we just use all the points?

File: omnigibson/prims/rigid_prim.py [hunk 5]
```diff
@@ -605,6 +605,142 @@ def contact_reporting_enabled(self):
...
+        """
+        return self._compute_points_on_convex_hull(visual=False)
+        
+    @property
+    def aabb(self):
+        position, orientation = self.get_position_orientation()
+        hull_points = self.collision_boundary_points
+        if hull_points is None:
+            # TODO: Decide if this is the right thing to do
+            return position, position
```
Comment (ChengshuLi) [inline omnigibson/prims/rigid_prim.py:655]: Add comments to explain when would this happen

File: omnigibson/prims/rigid_prim.py [hunk 6]
```diff
@@ -605,6 +605,142 @@ def contact_reporting_enabled(self):
...
+                # Append corners of the bounding box
+                points_to_transform.extend([
+                    [lo[0], lo[1], lo[2]], [hi[0], lo[1], lo[2]],
+                    [lo[0], hi[1], lo[2]], [hi[0], hi[1], lo[2]],
+                    [lo[0], lo[1], hi[2]], [hi[0], lo[1], hi[2]],
+                    [lo[0], hi[1], hi[2]], [hi[0], hi[1], hi[2]]
+                ])
+        else:
+            # Use predefined hull points directly
+            points_to_transform.extend(self.visual_boundary_points)
```
Comment (ChengshuLi) [inline omnigibson/prims/rigid_prim.py:709]: like mentioned before, we should just use all the points. By the way, the logic between visual_aabb and collision_aabb can be shared.

File: omnigibson/systems/macro_particle_system.py [hunk 7]
```diff
@@ -371,7 +371,7 @@ def update(cls):
         # Run super first
         super().update()
 
-        z_extent = cls.particle_object.aabb_extent[2]
+        z_extent = cls.particle_object.extent[2]
```
Comment (ChengshuLi) [inline omnigibson/systems/macro_particle_system.py:374]: maybe we need to asset scale is uniform before this.

File: omnigibson/object_states/particle_modifier.py [hunk 8]
```diff
@@ -1120,7 +1124,7 @@ def _modify_particles(self, system):
                     system.create_attachment_group(obj=self.obj)
                 avg_scale = np.cbrt(np.product(self.obj.scale))
                 scales = system.sample_scales_by_group(group=group, n=len(start_points))
-                cuboid_dimensions = scales * system.particle_object.aabb_extent.reshape(1, 3) * avg_scale
+                cuboid_dimensions = scales * system.particle_object.extent.reshape(1, 3) * avg_scale
```
Comment (ChengshuLi) [inline omnigibson/object_states/particle_modifier.py:1127]: also need to asset scale is uniform

File: omnigibson/systems/macro_particle_system.py [hunk 9]
```diff
@@ -389,6 +389,7 @@ def update(cls):
                 z_up = np.zeros_like(normals)
                 z_up[:, 2] = 1.0
                 orientations = T.axisangle2quat(T.vecs2axisangle(z_up, normals))
+                z_extent = cls._particle_object.extent[2]
```
Comment (ChengshuLi) [inline omnigibson/systems/macro_particle_system.py:392]: no need to compute again?

File: omnigibson/utils/geometry_utils.py [hunk 10]
```diff
@@ -385,9 +385,12 @@ def calculate_volume(precision=1e-5):
             mesh.visible = True
 
         # Determine equally-spaced sampling distance to achieve this minimum particle count
-        aabb_volume = np.product(volume_link.aabb_extent)
+        try: 
+            aabb_volume = np.product(volume_link.visual_aabb_extent)
+        except NotImplementedError:
+            return 0
```
Comment (ChengshuLi) [inline omnigibson/utils/geometry_utils.py:391]: same, what edge case is this?

File: tests/test_transition_rules.py [hunk 11]
```diff
@@ -666,7 +665,7 @@ def test_cooking_system_rule_failure_nonrecipe_objects():
     rosemary = get_system("rosemary")
     chicken_soup = get_system("cooked__chicken_soup")
 
-    place_obj_on_floor_plane(stove)
+    place_obj_on_floor_plane(stove, y_offset=0.0077189438, z_offset=0.0051004850)
```
Comment (ChengshuLi) [inline tests/test_transition_rules.py:668]: are these changes actually needed?

**Comment by hang-yin** [comment]
Profiling on og-develop (with 1 Fetch in Rs_int: load time 46.1, fps 11.238):
profile_og_dev
Profiling on this branch (with 1 Fetch in Rs_int: load time 45.6, fps 11.678):
profile_new_aabb

File: omnigibson/prims/rigid_prim.py (hunk 6 above)
Comment (ChengshuLi) [reply omnigibson/prims/rigid_prim.py]: will be fixed in https://github.com/StanfordVL/OmniGibson/pull/582

**Review by cgokmen — APPROVED** [review]
